plugin/PlayAssetDelivery/src/test/java/com/google/play/core/godot/assetpacks/PlayAssetDeliveryTest.java

```diff
@@ -189,8 +196,77 @@ public void getPackLocations_success() {
   }
 
   @Test
-  public void remove_success() {
-    // Mock the side effects of removePackSuccessTaskMock object, call onSuccessListener the instant
+  public void getPackStates_success() {
+    // Mock the side effects of Task<AssetPackStates> object, call onSuccessListener the instant
+    // it is registered.
+    Dictionary testDict = PlayAssetDeliveryTestHelper.createAssetPackStatesTestDictionary();
+    AssetPackStates testAssetPackStates = new AssetPackStatesFromDictionary(testDict);
+
+    doAnswer(
```

Comment:
Since we are reusing the TaskMock object, it is possible to refactor this setup part to an one-time-only-call using @Before tag. The downside of this is that it will make each individual test case less self-contained, especially when we need to declare separate private variable like ```testAssetPackStates``` so that both the closure and this test function could reference it.

Reply:
That's a good point. I wonder if we could have a mockTaskFactory with generic methods to produce tasks that are mocked to produce a specific value:
````
public static Task<T> createMockOnSuccessTask(T result)
```` 

That way we can remove a bunch of the setup code in the individual tests.

Reply:
Refactored duplicate code into factory methods in PlayAssetDeliveryTestHelper class.